pkg/agent/agent_test.go

```diff
@@ -195,6 +205,46 @@ func TestInitNodeLocalConfig(t *testing.T) {
 			expectedMTU:            1400,
 			expectedNodeAnnotation: nil,
 		},
+		{
+			name:             "noencap mode",
+			trafficEncapMode: config.TrafficEncapModeNoEncap,
+			mtu:              0,
+			expectedMTU:      1500,
+			expectedNodeAnnotation: map[string]string{
+				types.NodeMACAddressAnnotationKey:       transportIfaceMAC.String(),
+				types.NodeTransportAddressAnnotationKey: transportIP.String(),
```

Comment:
It's strange that this case has exactly same input as the first case but expects different output.
I see you are actually changing input based on expected output when testing, which is a bit obscure.
You can just add transportIface to input:
```
	tests := []struct {
		name                   string
		trafficEncapMode       config.TrafficEncapModeType
		tunnelType             ovsconfig.TunnelType
                transportInterface     string
		mtu                    int
		expectedMTU            int
		expectedNodeAnnotation map[string]string
	}{
		{
			name:                   "noencap mode",
			trafficEncapMode:       config.TrafficEncapModeNoEncap,
			mtu:                    0,
			expectedMTU:            1500,
			expectedNodeAnnotation: map[string]string{types.NodeMACAddressAnnotationKey: macAddr.String()},
		},
		{
			name:                   "noencap mode with transportInterface",
			trafficEncapMode:       config.TrafficEncapModeNoEncap,
			mtu:                    0,
                        transportInterface: transportIfaceName,
			expectedMTU:            1500,
			expectedNodeAnnotation: map[string]string{
				types.NodeMACAddressAnnotationKey:       transportIfaceMAC.String(),
				types.NodeTransportAddressAnnotationKey: transportIP.String(),
                         },
		},
```
And subtests should have different names to differentiate failures.